res/res_pjsip_stir_shaken.c

```diff
@@ -510,17 +513,27 @@ static void stir_shaken_outgoing_request(struct ast_sip_session *session, pjsip_
 		return;
 	}
 
-	if (ast_strlen_zero(session->id.number.str) && session->id.number.valid) {
+	ast_party_id_init(&connected_id);
+	ast_channel_lock(session->channel);
+	effective_id = ast_channel_connected_effective_id(session->channel);
+	ast_party_id_copy(&connected_id, &effective_id);
+	ast_channel_unlock(session->channel);
+
+	/* XXX: This if statement is logically the same as the first if statement in
+	   res_pjsip_caller_id.c:add_id_headers. They should be combined. */
+	if (!connected_id.number.valid
+		|| (!session->endpoint->id.trust_outbound
+			&& (ast_party_id_presentation(&connected_id) & AST_PRES_RESTRICTION) != AST_PRES_ALLOWED)) {
+		ast_party_id_free(&connected_id);
 		return;
```

Comment:
It would be better to do what this comment says than to kick it down the road.

Reply:
Can you clarify why "combined" means within this context? Having the STIR/SHAKEN logic be in callerid, for example?